kore/src/Kore/Step/Simplification/Ceil.hs

```diff
@@ -313,6 +319,34 @@ makeEvaluateNormalizedAc
             ++ elementsWithVariablesDistinct
     And.simplifyEvaluatedMultiPredicate (MultiAnd.make allConditions)
   where
+    assertCanBeEqCheck
+        :: TermLike variable
+        -> simplifier (TermLike variable)
+    assertCanBeEqCheck key =
+        case key of
+            App_ symbol _ ->
+                if not (Symbol.isInjective symbol)
```

Comment:
I know that I said differently in the past (sorry for the confusion), but:
1. I think that having constructors at the top is not enough, since the keys can be, say, `C(f())` and `C(g())`, where `C` is a constructor, and `f` and `g` are uninterpreted symbols, and then there's no way of comparing them.
2. I don't think all injections work, only sort injections do, since you can't compare random distinct injections.

3. @ttuegel I think sort injection do not work unless they are simplified, since we can transform `inj(inj)` into a single `inj`. Should we assume that? If yes, should we check it?

Reply:
> I think sort injection do not work unless they are simplified, since we can transform `inj(inj)` into a single `inj`. Should we assume that? If yes, should we check it?

We should not assume that sort injections are non-simplifiable, but we should inspect the child to see if it is a sort injection. We did this in the past, but I don't know what happened.